scrape/scrape.go

```diff
@@ -720,15 +721,24 @@ func resolveConflictingExposedLabels(lb *labels.Builder, exposedLabels, targetLa
 	}
 }
 
+func labelSliceHas(lbls []labels.Label, name string) bool {
+	for _, l := range lbls {
+		if l.Name == name {
+			return true
+		}
+	}
```

Comment:
does it make sense to assume sorted and do something like `if l.Name > name {return false}` ?

Reply:
I think not: we know `conflictingExposedLabels` is sorted at the beginning, but after we have renamed one `foo` to `exported_foo` it is no longer certain.